cs/test/core/Util.cs

```diff
@@ -271,6 +273,22 @@ public static ArraySegment<byte> SerializeUnsafeCB<T>(T obj)
             return output.Data;
         }
 
+        public static ArraySegment<byte> SerializeUnsafeCBWithPtr<T>(T obj)
```

Comment:
Depends. If we want to be memory conservative in the tests yes. However, there is little gain and more complexity on the test side. After all we are treating an array as its ptr representation (ownership should not be transfered to the OutputPtrBuffer).